Running updates with policies with v3.221.0 fails

### What happened?

Running an update using v3.221.0 and running the policy that has been downloaded from Pulumi Cloud gives this error message:

```console
Diagnostics:
  pulumi:pulumi:Stack (k8s-container-ts-9-dev):
    Error: Cannot find module '@grpc/grpc-js'
    Require stack:
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-google-cloud-v2.0.3/node_modules/@pulumi/pulumi/runtime/settings.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-google-cloud-v2.0.3/node_modules/@pulumi/pulumi/log/index.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-google-cloud-v2.0.3/node_modules/@pulumi/pulumi/tsutils.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-google-cloud-v2.0.3/node_modules/@pulumi/pulumi/cmd/run-policy-pack/run.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-google-cloud-v2.0.3/node_modules/@pulumi/pulumi/cmd/run-policy-pack/index.js
        at Module._resolveFilename (node:internal/modules/cjs/loader:1448:15)
        at defaultResolveImpl (node:internal/modules/cjs/loader:1059:19)
        at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1064:22)
        at Module._load (node:internal/modules/cjs/loader:1234:25)
        at TracingChannel.traceSync (node:diagnostics_channel:328:14)
        at wrapModuleLoad (node:internal/modules/cjs/loader:245:24)
        at Module.require (node:internal/modules/cjs/loader:1548:12)
        at require (node:internal/modules/helpers:152:16)
        at Object.<anonymous> (/root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-google-cloud-v2.0.3/node_modules/@pulumi/pulumi/runtime/settings.js:23:27)
        at Module._compile (node:internal/modules/cjs/loader:1804:14)
    Error: Cannot find module '@grpc/grpc-js'
    Require stack:
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-aws-v2.6.3/node_modules/@pulumi/pulumi/runtime/settings.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-aws-v2.6.3/node_modules/@pulumi/pulumi/log/index.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-aws-v2.6.3/node_modules/@pulumi/pulumi/tsutils.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-aws-v2.6.3/node_modules/@pulumi/pulumi/cmd/run-policy-pack/run.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-aws-v2.6.3/node_modules/@pulumi/pulumi/cmd/run-policy-pack/index.js
        at Module._resolveFilename (node:internal/modules/cjs/loader:1448:15)
        at defaultResolveImpl (node:internal/modules/cjs/loader:1059:19)
        at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1064:22)
        at Module._load (node:internal/modules/cjs/loader:1234:25)
        at TracingChannel.traceSync (node:diagnostics_channel:328:14)
        at wrapModuleLoad (node:internal/modules/cjs/loader:245:24)
        at Module.require (node:internal/modules/cjs/loader:1548:12)
        at require (node:internal/modules/helpers:152:16)
        at Object.<anonymous> (/root/.pulumi/policies/pequod/pulumi-analyzer-hitrust-aws-v2.6.3/node_modules/@pulumi/pulumi/runtime/settings.js:23:27)
        at Module._compile (node:internal/modules/cjs/loader:1804:14)
    error: [runtime] Running program '/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2' failed with an unhandled exception:
        Error: Cannot find module '@pulumi/aws'
    Require stack:
    - /root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2/index.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2/node_modules/@pulumi/pulumi/cmd/run-policy-pack/run.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2/node_modules/@pulumi/pulumi/cmd/run-policy-pack/index.js
        at Module.<anonymous> (node:internal/modules/cjs/loader:1448:15)
        at Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
        at defaultResolveImpl (node:internal/modules/cjs/loader:1059:19)
        at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1064:22)
        at Module._load (node:internal/modules/cjs/loader:1234:25)
        at TracingChannel.traceSync (node:diagnostics_channel:328:14)
        at wrapModuleLoad (node:internal/modules/cjs/loader:245:24)
        at Module.require (node:internal/modules/cjs/loader:1548:12)
        at require (node:internal/modules/helpers:152:16)
        at Object.<anonymous> (/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2/index.js:36:11) {
      code: 'MODULE_NOT_FOUND',
      requireStack: [
        '/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2/index.js',
        '/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2/node_modules/@pulumi/pulumi/cmd/run-policy-pack/run.js',
        '/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-aws-v1.3.2/node_modules/@pulumi/pulumi/cmd/run-policy-pack/index.js'
      ]
    }
    error: [runtime] Running program '/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2' failed with an unhandled exception:
        Error: Cannot find module '@pulumi/gcp'
    Require stack:
    - /root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2/index.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2/node_modules/@pulumi/pulumi/cmd/run-policy-pack/run.js
    - /root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2/node_modules/@pulumi/pulumi/cmd/run-policy-pack/index.js
        at Module.<anonymous> (node:internal/modules/cjs/loader:1448:15)
        at Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
        at defaultResolveImpl (node:internal/modules/cjs/loader:1059:19)
        at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1064:22)
        at Module._load (node:internal/modules/cjs/loader:1234:25)
        at TracingChannel.traceSync (node:diagnostics_channel:328:14)
        at wrapModuleLoad (node:internal/modules/cjs/loader:245:24)
        at Module.require (node:internal/modules/cjs/loader:1548:12)
        at require (node:internal/modules/helpers:152:16)
        at Object.<anonymous> (/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2/index.js:36:11) {
      code: 'MODULE_NOT_FOUND',
      requireStack: [
        '/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2/index.js',
        '/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2/node_modules/@pulumi/pulumi/cmd/run-policy-pack/run.js',
        '/root/.pulumi/policies/pequod/pulumi-analyzer-pulumi-best-practices-google-cloud-v1.0.2/node_modules/@pulumi/pulumi/cmd/run-policy-pack/index.js'
      ]
    }
    (node:767) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
    (Use `node --trace-deprecation ...` to show where the warning was created)
```

### Example

see above for error message

### Output of `pulumi about`

n/a

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/changelog/pending/20260219--engine--report-policy-pack-installation-errors.yaml
+++ b/changelog/pending/20260219--engine--report-policy-pack-installation-errors.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: fix
+  scope: engine
+  description: Report policy pack installation errors

--- a/pkg/engine/lifecycletest/analyzer_test.go
+++ b/pkg/engine/lifecycletest/analyzer_test.go
@@ -15,6 +15,7 @@
 package lifecycletest
 
 import (
+	"bytes"
 	"context"
 	"encoding/json"
 	"errors"
@@ -1170,4 +1171,112 @@ func TestAnalyzeStackRunsInParallel(t *testing.T) {
 	close(gate)
 
 	require.NoError(t, <-done)
+}
+
+// failingDownloadRequiredPolicy is a RequiredPolicy that fails during download, used to test that
+// policy pack installation errors are properly surfaced rather than silently dropped.
+type failingDownloadRequiredPolicy struct {
+	name string
+}
+
+func (p *failingDownloadRequiredPolicy) Name() string                        { return p.name }
+func (p *failingDownloadRequiredPolicy) Version() string                     { return "" }
+func (p *failingDownloadRequiredPolicy) Installed() bool                     { return false }
+func (p *failingDownloadRequiredPolicy) LocalPath() (string, error)          { return "", nil }
+func (p *failingDownloadRequiredPolicy) Config() map[string]*json.RawMessage { return nil }
+func (p *failingDownloadRequiredPolicy) Download(
+	_ context.Context,
+	_ func(io.ReadCloser, int64) io.ReadCloser,
+) (io.ReadCloser, int64, error) {
+	return nil, 0, errors.New("policy pack download failed")
+}
+
+func (p *failingDownloadRequiredPolicy) Install(_ *plugin.Context, _ io.ReadCloser, _, _ io.Writer) error {
+	return nil
+}
+
+// failingInstallRequiredPolicy is a RequiredPolicy that fails during installation, used to test that
+// policy pack installation errors are properly surfaced rather than silently dropped.
+type failingInstallRequiredPolicy struct {
+	name string
+}
+
+func (p *failingInstallRequiredPolicy) Name() string                        { return p.name }
+func (p *failingInstallRequiredPolicy) Version() string                     { return "" }
+func (p *failingInstallRequiredPolicy) Installed() bool                     { return false }
+func (p *failingInstallRequiredPolicy) LocalPath() (string, error)          { return "", nil }
+func (p *failingInstallRequiredPolicy) Config() map[string]*json.RawMessage { return nil }
+func (p *failingInstallRequiredPolicy) Download(
+	_ context.Context,
+	_ func(io.ReadCloser, int64) io.ReadCloser,
+) (io.ReadCloser, int64, error) {
+	return io.NopCloser(bytes.NewReader(nil)), 0, nil
+}
+
+func (p *failingInstallRequiredPolicy) Install(_ *plugin.Context, _ io.ReadCloser, _, _ io.Writer) error {
+	return errors.New("policy pack install failed")
+}
+
+// TestPolicyPackDownloadFailureReturnsError is a regression test verifying that errors from the download
+// step of policy pack installation are returned to the caller rather than silently dropped.
+func TestPolicyPackDownloadFailureReturnsError(t *testing.T) {
+	t.Parallel()
+
+	loaders := []*deploytest.PluginLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{}, nil
+		}),
+	}
+
+	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, _ *deploytest.ResourceMonitor) error {
+		return nil
+	})
+	hostF := deploytest.NewPluginHostF(nil, nil, programF, loaders...)
+
+	p := &lt.TestPlan{
+		Options: lt.TestUpdateOptions{
+			T:                t,
+			SkipDisplayTests: true,
+			UpdateOptions: UpdateOptions{
+				RequiredPolicies: []RequiredPolicy{&failingDownloadRequiredPolicy{name: "failing-download-policy"}},
+			},
+			HostF: hostF,
+		},
+	}
+
+	project := p.GetProject()
+	_, err := lt.TestOp(Update).Run(project, p.GetTarget(t, nil), p.Options, false, p.BackendClient, nil)
+	require.ErrorContains(t, err, "policy pack download failed")
+}
+
+// TestPolicyPackInstallFailureReturnsError is a regression test verifying that errors from the install
+// step of policy pack installation are returned to the caller rather than silently dropped.
+func TestPolicyPackInstallFailureReturnsError(t *testing.T) {
+	t.Parallel()
+
+	loaders := []*deploytest.PluginLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{}, nil
+		}),
+	}
+
+	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, _ *deploytest.ResourceMonitor) error {
+		return nil
+	})
+	hostF := deploytest.NewPluginHostF(nil, nil, programF, loaders...)
+
+	p := &lt.TestPlan{
+		Options: lt.TestUpdateOptions{
+			T:                t,
+			SkipDisplayTests: true,
+			UpdateOptions: UpdateOptions{
+				RequiredPolicies: []RequiredPolicy{&failingInstallRequiredPolicy{name: "failing-install-policy"}},
+			},
+			HostF: hostF,
+		},
+	}
+
+	project := p.GetProject()
+	_, err := lt.TestOp(Update).Run(project, p.GetTarget(t, nil), p.Options, false, p.BackendClient, nil)
+	require.ErrorContains(t, err, "policy pack install failed")
 }

--- a/pkg/engine/update.go
+++ b/pkg/engine/update.go
@@ -488,11 +488,11 @@ func newUpdateSource(ctx context.Context,
 	client deploy.BackendClient, opts *deploymentOptions, proj *workspace.Project, pwd, main, projectRoot string,
 	target *deploy.Target, plugctx *plugin.Context, resourceHooks *deploy.ResourceHooks, panicErrs chan<- error,
 ) (deploy.Source, error) {
-	manager := newInstallManager(false /*returnPluginErrors*/)
-
 	//
 	// Step 1: Install policy packs and plugins.
 	//
+
+	manager := newInstallManager(false /*returnPluginErrors*/)
 
 	ensurePoliciesAreInstalled(ctx, plugctx, opts, opts.RequiredPolicies, manager)
 
@@ -512,7 +512,7 @@ func newUpdateSource(ctx context.Context,
 	}
 
 	if err := manager.Wait(); err != nil {
-		logging.V(7).Infof("newUpdateSource(): failed to install policy packs and plugins: %v", err)
+		return nil, err
 	}
 
 	//

